presto-main/src/test/java/com/facebook/presto/operator/scalar/TestUnitFunctions.java

```diff
@@ -0,0 +1,105 @@
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+
+import static io.airlift.units.DataSize.Unit.BYTE;
+import static io.airlift.units.DataSize.Unit.GIGABYTE;
+import static io.airlift.units.DataSize.Unit.KILOBYTE;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static io.airlift.units.DataSize.Unit.PETABYTE;
+import static io.airlift.units.DataSize.Unit.TERABYTE;
+import static io.airlift.units.Duration.succinctDuration;
+import static io.airlift.units.Duration.succinctNanos;
+import static java.util.concurrent.TimeUnit.DAYS;
+import static java.util.concurrent.TimeUnit.HOURS;
+import static java.util.concurrent.TimeUnit.MICROSECONDS;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class TestUnitFunctions
+        extends AbstractTestFunctions
+{
+    private void testDuration(String sqlExpr, Duration expected)
+    {
+        assertFunction(sqlExpr, VarcharType.VARCHAR, expected.toString());
+    }
+
+    @Test
+    public void testSuccinctDuration()
+    {
+        testDuration("succinct_duration(123, 'ns')", succinctNanos(123));
+        testDuration("succinct_duration(123.45, 'ns')", succinctDuration(123.45, NANOSECONDS));
+        testDuration("succinct_duration(123.45, 'us')", succinctDuration(123.45, MICROSECONDS));
+        testDuration("succinct_duration(123.45, 'ms')", succinctDuration(123.45, MILLISECONDS));
+        testDuration("succinct_duration(123.45, 's')", succinctDuration(123.45, SECONDS));
+        testDuration("succinct_duration(12.34, 'm')", succinctDuration(12.34, MINUTES));
+        testDuration("succinct_duration(12.34, 'h')", succinctDuration(12.34, HOURS));
+        testDuration("succinct_duration(12.34, 'd')", succinctDuration(12.34, DAYS));
+        testDuration("succinct_duration(3600, 's')", new Duration(1.00, HOURS));
+
+        testDuration("succinct_nanos(123)", succinctNanos(123));
+        testDuration("succinct_nanos(123.4)", succinctDuration(123.4, NANOSECONDS));
+        testDuration("succinct_nanos(123456)", new Duration(123.456, MICROSECONDS));
+        testDuration("succinct_millis(123)", succinctDuration(123, MILLISECONDS));
+        testDuration("succinct_millis(123.4)", succinctDuration(123.4, MILLISECONDS));
+        testDuration("succinct_millis(1234)", new Duration(1.234, SECONDS));
+
+        assertInvalidFunction("succinct_duration(-1, 'ns')", INVALID_FUNCTION_ARGUMENT);
+        assertInvalidFunction("succinct_duration(123, 'sec')", INVALID_FUNCTION_ARGUMENT);
+        assertInvalidFunction("succinct_nanos(-123)", INVALID_FUNCTION_ARGUMENT);
+        assertInvalidFunction("succinct_millis(-123)", INVALID_FUNCTION_ARGUMENT);
```

Comment:
please add test cases for `succinct_nanos` for infinity and nan (and other functions with double input such as `succinct_millis` etc.)